src/main/java/pl/javastart/task/ReadFromUser.java

```diff
@@ -0,0 +1,101 @@
+package pl.javastart.task;
+
+import java.io.*;
+import java.util.PriorityQueue;
+import java.util.Queue;
+import java.util.Scanner;
+
+public class ReadFromUser {
+    private Queue<Vehicle> vehicles = new PriorityQueue<>();
+    private Scanner scanner = new Scanner(System.in);
+
+    void mainLoop() {
+        Option option;
+        do {
+            System.out.println("Wybierz opcje:");
+            printOptions();
+            option = Option.createFromInt(scanner.nextInt());
+            scanner.nextLine();
+            if (option == Option.ADD) {
+                vehicles.add(addNewVehicle());
+            } else if (option == Option.TAKE) {
+                if (vehicles.isEmpty()) {
+                    try {
+                        readVehiclesFromFile(vehicles, "vehicles.csv");
+                    } catch (FileNotFoundException e) {
+                        System.err.println("Nie udalo sie odczytac danych z pliku");
+                    }
+                } else {
+                    takeVehicle();
+                }
+            } else if (option == Option.EXIT) {
+                System.out.println("Koniec programu!");
+            }
+            if (!vehicles.isEmpty()) {
+                try {
+                    saveVehiclesToFile("vehicles.csv");
+                } catch (IOException e) {
+                    System.err.println("Nie udało się zapisać danych do pliku.");
+                }
+            }
```

Comment:
Jeżeli kolejka będzie pusta, to też warto zapisać ją w pliku, żeby przy ponownym uruchomieniu nie przeglądać tych samych pojazdów